### PR TITLE
[cxx-interop] Enable hardening when running tests for std::span

### PR DESCRIPTION
**Pull request by susmonteiro**

### CODE REVIEW

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

**Comment by susmonteiro** [comment]
@swift-ci please smoke test

File: test/Interop/Cxx/stdlib/use-std-span.swift [hunk 1]
```diff
@@ -13,6 +14,24 @@ import CxxStdlib
 
 var StdSpanTestSuite = TestSuite("StdSpan")
 
+func takesSpanOfInt(s: Span) {
```
Comment (egorzhdan) [inline test/Interop/Cxx/stdlib/use-std-span.swift:17]: Minor: we usually don't use argument labels for function parameters in these test utilities, unless they help with understanding of the semantics of the utility. In this case you could spell the function signature as `func takesSpanOfInt(_ s: Span)` to omit the argument label.

**Review by egorzhdan — COMMENTED** [review]
Thanks! Just one tiny comment.

File: test/Interop/Cxx/stdlib/use-std-span.swift (hunk 1 above)
Comment (susmonteiro) [reply test/Interop/Cxx/stdlib/use-std-span.swift:17]: Sure! I'll fix this

**Review by egorzhdan — APPROVED** [review]
LGTM!

**Pull request merged**